### DISCUSSION TITLE
Parcel inserts bundled CSS immediately after <head>, so it is overwritten by other components.

### DISCUSSION THREAD

**Discussion by Cristy94** [parcel-bundler/parcel#4820] [General]
I am using `"parcel": "^2.0.0-nightly.318",` and the bundled CSS is imported immediately after `<head>`, but I need it to be imported last so that my CSS should overwrite all other global CSS included.

Was this changed recently?
Why does it import at the top of `<head>` instead of just before `</head>`?

**Comment by Cristy94** [comment]
Any suggestions on how I can make the bundled CSS be added later in the `<head>` tag, not as first child?

**Comment by mischnic** [comment]
Could you give an example?

**Comment by Cristy94** [comment]
Sure,

So my **index.html** looks like this before build:

```HTML
...
<head>
<link rel="stylesheet" href="www.example.com/some.css" />
</head>
...
```

And after build it looks like this:

```HTML
...
<head>
<link rel="stylesheet" href="/bundle.css" />
<link rel="stylesheet" href="www.example.com/some.css" />
</head>
...
```

But I want it to look like this:

```HTML
...
<head>
<link rel="stylesheet" href="www.example.com/some.css" />
<link rel="stylesheet" href="/bundle.css" /> <!-- THIS SHOULD BE LAST, SO IT OVERWRITES OTHER RULES -->
</head>
...
```

The main issue is that I use MaterialUI in my project and I need to have this order:

- material-icons.css
- MUI inline injected CSS
- bundle.css

I can currently do this, but it's not what I want:

- MUI inline injected CSS (I inject this as the first child of <head>)
- bundle.css (this was automatically added at the top)
- material-icons.css (problem is that this now overwrites MaterialUI font sizes for icons and also whatever I write in my CSS bundle).

So, how can I make sure that the parcel **bundle.css** will be included after the other stylesheets imported in **index.html**?

**Comment by Cristy94** [comment]
@mischnic Any suggestions? 🙏

**Comment by mischnic** [comment]
So with
```html
<head>
	<link rel="stylesheet" href="https://www.example.com/some.css" />
	<script src="index.js"></script>
</head>
```
```css
body {
	color: red;
}
```
```js
import "./index.css";
```
I get:
```html
<head>
	<link rel="stylesheet" href="https://www.example.com/some.css" />
	<link rel="stylesheet" href="/index.2c4451d2.css" />
	<script src="/index.f81b525c.js"></script>
</head>
```

**Comment by Cristy94** [comment]


**Reply by Cristy94** [reply]
Also the script generated is added at the end of the `</body>` tag, not inside `<head>`. Oh, that is because I have `<script src="./index.tsx"></script>` after the `<div id="app"></div>`. But still can't explain the CSS at the top.

**Comment by Cristy94** [comment]
I found the issue! Whenever you add an inline style in the head, the CSS is imported at the top:

Try this:

```
<head>
	<link rel="stylesheet" href="https://www.example.com/some.css" />
        <style>html,body {background: #e9eaed;}</style> <!-- ADD THIS -->
</head>

```

The output now looks like:
```

<head><link rel="stylesheet" href="/index.22506e2f.css">
  <link rel="stylesheet" href="https://www.example.com/some.css">
  <style>html,body {background: #e9eaed;}</style>
</head>
```

Any idea why simply adding a `<style>` tag makes the bundled CSS to be placed at the top? It doesn't even add a new line before it anymore (looks bugged?).

**Comment by Cristy94** [comment]
Created an issue for this: https://github.com/parcel-bundler/parcel/issues/4882
